compiler/haskell-ide-core/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -67,6 +75,27 @@ import Language.Haskell.LSP.Diagnostics
 
 import Development.IDE.Types.Location
 
+-- | Newtype wrapper around FilePath that always has normalized slashes.
+newtype NormalizedFilePath = NormalizedFilePath FilePath
+    deriving (Eq, Ord, Show, Hashable, NFData)
+
+instance IsString NormalizedFilePath where
+    fromString = toNormalizedFilePath
+
+toNormalizedFilePath :: FilePath -> NormalizedFilePath
+toNormalizedFilePath "" = NormalizedFilePath ""
+toNormalizedFilePath fp = NormalizedFilePath $ normalise' fp
```

Comment:
If all we want to do is replace `\` with `/` I'd be tempted by `replace "\\" "/"` since I worry the drive/path stuff is scary and more likely to go wrong in corner cases (and that's 90% because I wrote all those functions).

Reply:
Changed but we only replace on Windows since backslashes are valid in filenames on Linux.